### PR TITLE
Added build job

### LINKED ISSUE #96: [Feature]: Add build CI job
### To which use case does your feature request relate?

_No response_

### Is your feature request related to a problem? Please describe

Currently, we don't have a build job in the CI. Uploading the build artifact may be helpful and will be required when finishing a release later on.

### Describe the solution you'd like

Add a CI build job, which uses the build command in the package.json and upload the created artifact.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Slartibartfass2**
Closes #96

## What I have made

- added a build job that uploads the build artifact
- added a lighthouse job that uses the build artifact to analyze the pages
  - this a long-running job, I wouldn't make it required
  - maybe I would only let it run when something is merged to develop or main
  - in the future we could deploy the reports e.g. using GitHub Pages

## Checklist

Either tick or cross out the items that do not apply (using \~\~example text\~\~) and give a reason why the item does not apply.

- [ ] ~~I have updated the documentation accordingly and commented my code~~ (only CI jobs)
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~ (only CI jobs)
- [ ] ~~(for reviewer) I have checked the implementation against the requirements~~ (only CI jobs)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+            - name: Archive build artifacts
+              uses: actions/upload-artifact@v4
+              if: ${{ !cancelled() }}
+              with:
+                  name: build-artifact
+                  path: .svelte-kit/output/
+                  if-no-files-found: error
+
+    lighthouse:
```
Comment (Merseleo) [inline .github/workflows/build.yml]: Maybe only run this job, [wenn the pull request was approved](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved), which should happen before a merge to develop / main, but no block the ci all the time.

File: .github/lighthouse.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export default {
+    extends: "lighthouse:default",
+    settings: {
+        formFactor: "desktop",
+        screenEmulation: {
+            mobile: false,
+            width: 1440,
+            height: 1024,
```
Comment (Merseleo) [inline .github/lighthouse.config.js]: I know we used this resolution in Figma, but why don't we check for the [most common resolution](https://www.browserstack.com/guide/ideal-screen-sizes-for-responsive-design): 1920 x 1080

**Review by Merseleo — CHANGES_REQUESTED** [review]
Imo it looks good for now, maybe we want to extend the assertions etc. for Lighthouse later, but let's start with these jobs. By the way a nice idea to build the frontend and check it with Lighthouse.

File: .github/lighthouse.config.js (hunk 2 above)
Comment (Slartibartfass2) [reply .github/lighthouse.config.js]: We can use this too, as long as it's a fixed size.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Slartibartfass2) [reply .github/workflows/build.yml]: This seems too late in my opinion. When the PR is approved, there's no reason to see the results of this job.

I discussed this quick with @luca-schlecker, and we think it's best to just do it when something was pushed to main or develop. The developer/reviewer can check the report manually on their own machine.

**Review by Merseleo — COMMENTED** [review]
Ok then let's run the lighthouse job on a requested review.

**Review by Merseleo — COMMENTED** [review]
I think we are done with it.

**Review by Merseleo — APPROVED** [review]
Seems to work now, so let's merge ...

**Pull request merged**